buildtest/tools/modulesystem/tree.py

```diff
@@ -5,25 +5,6 @@
 from buildtest.tools.modules import update_spider_file
 
 
-def func_module_tree_subcmd(args):
-    """ Entry point for ``buildtest module tree`` subcommand """
```

Comment:
How do we expect buildtest to update spider file when they change BUILDTEST_MODULEPATH. The idea was for user to not update that variable from config file. But if they do from config file we need a history to check what was the value before they update the file. I guess we could capture this detail in some json file maybe even The spider file. The spider command needs to run against all module trees. That's the essence of getting all the module features working

Reply:
So - let's step back and discuss what the purpose of this spider file is. You want to always keep a total representation of all system modules? Why? The module path is logical to keep if we need to look for modules, but keeping the listing cached is a huge responsibility, and in that the module is a kind of environment, and one that has a specific state at the run of any recipe, I don't think it makes sense to try to always keep track of this. To play devil's advocate, let's say we do keep track of it. Then the user is required to run that command. Are they really going to do that? Probably not. It makes much more sense to interact with spider and derive some state when it's actually needed, for doing a build test.